src/backend/distributed/sql/udfs/citus_stat_activity/latest.sql

```diff
@@ -0,0 +1,13 @@
+DROP VIEW IF EXISTS pg_catalog.citus_stat_activity;
+
+CREATE VIEW citus.citus_stat_activity AS
+SELECT global_pid, nodeid, pg_stat_activity_from_all_nodes.* FROM
+run_command_on_workers($$
+    SELECT to_json(array_agg(pg_stat_activity.*)) FROM pg_stat_activity
+$$) as text_result_from_all_nodes
+JOIN LATERAL json_populate_recordset(NULL::pg_stat_activity, text_result_from_all_nodes.result::JSON) as pg_stat_activity_from_all_nodes ON TRUE
+LEFT JOIN pg_dist_node ON text_result_from_all_nodes.nodename = pg_dist_node.nodename AND text_result_from_all_nodes.nodeport = pg_dist_node.nodeport
+LEFT JOIN get_global_active_transactions() ON pid = process_id AND initiator_node_identifier = pg_dist_node.nodeid;
```

Comment:
hmm, do we set `initiator_node_identifier` properly in get_global_active_transactions?

It is currently set: `		initiatorNodeIdentifier = currentBackend->citusBackend.initiatorNodeIdentifier;`


But, we should extract it from gpid?

Reply:
Then let me create the `extract_node_id_from_gpid` udf

Reply:
This turned out to be an incorrect approach. Because we are interested in the node id of the node the query runs on, not the initiator. But both `initiator_node_indentifier` and `extract_node_id_from_global_pid(global_pid)` are initiator node ids.